crates/project/src/dap_store.rs

```diff
@@ -1569,6 +1576,20 @@ impl DapStore {
         })
     }
 
+    async fn handle_get_debugger_sessions(
+        this: Model<Self>,
+        _envelope: TypedEnvelope<proto::GetDebuggerSessions>,
+        mut cx: AsyncAppContext,
+    ) -> Result<()> {
+        this.update(&mut cx, |dap_store, cx| {
+            if let Some((_, _)) = dap_store.downstream_client {
```

Comment:
Why do we need to check here if we have a downstream client?

Reply:
Just a sanity check. I don’t think it will ever fail but just in guess theirs a future regression or something we don’t think about.

Reply:
Uhm but this code is never hit inside the if statement right? Because we should only end up inside this method on the non host clients, so there will never be a downstream client. Or am I missing something?

Reply:
Shoot you’re right!! I should’ve use upstream_client instead. I was up late last night and didn’t realize.

Reply:
All good, I believe this is already covered by the collab code that it's only called when you are actually inside a shared collab channel.

Reply:
The check was supposed to ensure that only the host would respond to a GetDebuggerSessions message. I wasn't sure if a message is propagated to all clients or just the host.

Reply:
I tested this and the original downstream_client was correct. We only want to send active sessions if we have downstream clients